src/ParameterPostProcessor.php

```diff
@@ -60,10 +56,23 @@ private function convertValues(array $values, string $prefix = '') : array
 
             $convertedValues[$prefix . $key] = $value;
             if (is_array($value)) {
-                $convertedValues += $this->convertValues($value, $prefix . $key . '.');
+                $convertedValues += $this->resolveNestedParameters($value, $prefix . $key . '.');
             }
         }
 
         return $convertedValues;
     }
+
+    private function getResolvedParameters(): ParameterBagInterface
+    {
+        $resolved = $this->resolveNestedParameters($this->parameters);
+        $bag = new ParameterBag($resolved);
+        try {
+            $bag->resolve();
```

Comment:
Hm, okay.
I've added the return within the try catch. 

I usually only use try/catch around calls which provide the exception and, as the return itself cannot throw an exception, I pass it at the end of the try/catch blocks.

Is there any good reason to write it like that? Or is it just a personal decision?

Reply:
One reason is that if you'll not be doing any more work after the try/catch, you can do any error handling following it, so that work doesn't need to be within a block. This aids in readability.

As an example:

```php
try {
    $bag->resolve();
    return;
} catch (Throwable $e) {
}

// $e is now in scope, and the only way we get here is if a throwable was caught
```